main.qml

```diff
@@ -399,6 +399,49 @@ ApplicationWindow {
         leftPanel.balanceLabelText = qsTr("Balance (#%1%2)").arg(currentWallet.currentSubaddressAccount).arg(accountLabel === "" ? "" : (" – " + accountLabel));
     }
 
+    function onUriHandler(uri){
+        if(uri.startsWith("monero://")){
+            var address = uri.substring("monero://".length);
+
+            var params = {}
+            if(address.length === 0) return;
+            var spl = address.split("?");
+
+            if(spl.length > 2) return;
+            if(spl.length >= 1) {
+                // parse additional params
+                address = spl[0];
+
+                if(spl.length === 2){
+                    spl.shift();
+                    var item = spl[0];
+
+                    var _spl = item.split("&");
+                    for (var param in _spl){
+                        var _item = _spl[param];
+                        if(!_item.indexOf("=") > 0) continue;
+
+                        var __spl = _item.split("=");
+                        if(__spl.length !== 2) continue;
+
+                        params[__spl[0]] = __spl[1];
+                    }
+                }
+            }
+
```

Comment:
I have read (e.g. [here](https://msdn.microsoft.com/en-us/windows/desktop/aa767914#MainContent) that on Windows it's possible to encounter "percent encoding" sometimes in the arguments in connection with custom URI schemes, like `%20` for blanks somewhere, e.g. in `tx_description`. Did you test whether this actually occurs, and if yes, do you care about this somewhere i.e. decode?